framework/wazuh/core/utils.py

```diff
@@ -968,6 +968,25 @@ def check_section(agents_regex, split_section):
         check_section(auth_section, split_section='</auth>')
 
 
+def check_indexer(data: str):
+    """Check if modifying the indexer configuration is allowed.
+
+    Parameters
+    ----------
+    data : str
+        Configuration file content.
+    """
+    blocked_configurations = configuration.api_conf['upload_configuration']
+
+    if not blocked_configurations['indexer']['allow']:
+        try:
+            indexer_section = re.compile(r"<indexer>(.*)</indexer>", flags=re.MULTILINE | re.DOTALL)
+            if len(indexer_section.findall(data)) > 0:
+                raise WazuhError(1130)
+        except IndexError:
+            pass
```

Comment:
```suggestion
        with contextlib.suppress(IndexError):
            indexer_section = re.compile(r"<indexer>(.*)</indexer>", flags=re.MULTILINE | re.DOTALL)
            if len(indexer_section.findall(data)) > 0:
                raise WazuhError(1130)
```

This will require importing `contextlib`.